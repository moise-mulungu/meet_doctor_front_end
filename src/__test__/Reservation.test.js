import React from 'react';
// import { ReactDOM } from 'react-dom';
import { render } from '@testing-library/react';
// import ListAllReservations from '../components/Pages/reservations/Reservation';
import renderer from 'react-test-renderer';


describe('ListAllReservations', () => {
  const ListAllReservations = () => {
    return (
      <section className="reservations-container">
        <h1>All Reservations</h1>
        <table className="reservations-table">
          <thead>
            <tr>
              <th>Doctor</th>
              <th>City</th>
              <th>Bill</th>
              <th>Date</th>
            </tr>
          </thead>
          <tbody className="reservations-body">
            <tr>
              <td>Doctor</td>
              <td>City</td>
              <td>Bill</td>
              <td>Date</td>
            </tr>
          </tbody>
        </table>
      </section>
    );
  }
    it('renders without crashing', () => {
        const reservations = renderer.create(<ListAllReservations />).toJSON();
        });
    it('renders correctly', () => {
        const tree = renderer.create(<ListAllReservations />).toJSON();

        expect(tree).toMatchSnapshot();

    }
    );

    it('renders a table', () => {
        const table = render(<ListAllReservations/>);
        const tableDOM = table.queryAllByTitle('table');
        expect(tableDOM.length).not.toBe(1);     
    }
    );

    it('renders a table header', () => {
        const header  = render(<ListAllReservations />);
        const tableHeader = header.getByText('All Reservations');
        expect(tableHeader).toHaveAccessibleName; 
    }
    );

    it('renders a table body', () => {
        const body = render(<ListAllReservations />);
        const tableBody = body.getByText('All Reservations');
        expect(tableBody).toHaveAccessibleName;
    }
    );

});