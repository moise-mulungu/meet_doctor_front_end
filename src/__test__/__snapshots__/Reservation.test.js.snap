// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListAllReservations renders correctly 1`] = `
<section
  className="reservations-container"
>
  <h1>
    All Reservations
  </h1>
  <table
    className="reservations-table"
  >
    <thead>
      <tr>
        <th>
          Doctor
        </th>
        <th>
          City
        </th>
        <th>
          Bill
        </th>
        <th>
          Date
        </th>
      </tr>
    </thead>
    <tbody
      className="reservations-body"
    >
      <tr>
        <td>
          Doctor
        </td>
        <td>
          City
        </td>
        <td>
          Bill
        </td>
        <td>
          Date
        </td>
      </tr>
    </tbody>
  </table>
</section>
`;
